unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/Tests/RPCSocialApiBridgeShould.cs

```diff
@@ -0,0 +1,234 @@
+
+            var rpcServer = new RpcServer<MockSocialServerContext>();
+            rpcServer.SetHandler((port, _, _) => { FriendshipsServiceCodeGen.RegisterService(port, friendshipsService); });
+
+            context = new MockSocialServerContext();
+            rpcServer.AttachTransport(server, context);
+
+            var matrixInitializationBridge = Substitute.For<IMatrixInitializationBridge>();
+
+            rpcSocialApiBridge = new RPCSocialApiBridge(matrixInitializationBridge, userProfileBridge, () => client);
+            rpcSocialApiBridge.Initialize();
+            yield return rpcSocialApiBridge.InitializeAsync(default(CancellationToken)).ToCoroutine();
+
+            matrixInitializationBridge.OnReceiveMatrixAccessToken += Raise.Event<Action<string>>(ACCESS_TOKEN);
+        }
+
+        [UnityTest]
+        public IEnumerator Initialize()
+        {
+            return UniTask.ToCoroutine(async () =>
+            {
+                var cancellationToken = new CancellationToken();
+                var friends = new Dictionary<string, UserStatus>();
+
+                context.userList = new List<Users>()
+                {
+                    new ()
+                    {
+                        Users_ =
+                        {
+                            new User() { Address = "addr1" },
+                            new User() { Address = "addr2" }
+                        }
+                    },
+                    new ()
+                    {
+                        Users_ =
+                        {
+                            new User() { Address = "addr3" },
+                            new User() { Address = "addr4" }
+                        }
+                    }
+                };
+
+                rpcSocialApiBridge.OnFriendAdded += friend => { friends.Add(friend.userId, friend); };
+
+                await rpcSocialApiBridge.GetInitializationInformationAsync(cancellationToken);
+
+                Assert.AreEqual(4, friends.Count);
+            });
+        }
+
+        [UnityTest]
+        public IEnumerator RequestFriendshipAsync()
+        {
+            return UniTask.ToCoroutine(async () =>
+            {
+                var cancellationToken = new CancellationToken();
+                var friendId = "Friend ID";
+                var message = "Hello friend";
+                var createdAt = 1234;
+
+                context.updateFriendshipResponse = new UpdateFriendshipResponse()
+                {
+                    Event = new FriendshipEventResponse()
+                    {
+                        Request = new RequestResponse()
+                        {
+                            User = new User()
+                            {
+                                Address = OWN_ID
+                            },
+                            Message = message,
+                            CreatedAt = createdAt
+                        }
+                    }
+                };
```

Comment:
You can improve readability of the test by creating an auxiliary method `CreateUpdateFriendshipResponse(string address, string message, int createdAt);`

Reply:
Created two functions for simplicity